### PR TITLE
Separate lockedMap into two separate parameters

### PR DESCRIPTION
**Pull request by bhollis**
The "locked map" has been whittled down to two pretty simple cases and locked mods are already on their own. This separates out pinnedItems and excludedItems and handles them separately.

### CODE REVIEW

File: src/app/loadout-builder/LoadoutBuilderItem.tsx [hunk 1]
```diff
@@ -1,46 +1,29 @@
...
-
   return (
     <DraggableInventoryItem item={item}>
       <ItemPopupTrigger item={item}>
         {(ref, onClick) => (
-          <div
-            className={clsx({
-              'excluded-item': locked?.some(
-                (p) => p.type === 'exclude' && p.item.index === item.index
-              ),
```
Comment (bhollis) [inline src/app/loadout-builder/LoadoutBuilderItem.tsx:34]: A class that wasn't defined, and a condition that can never be true

**Review by ryan-rushton — APPROVED** [review]
One weird edge case I came up with is that you can pin an item, then exclude it. In the end the exclusion wins out. Only thing I can think of is filter a pinned bucket in the excluded item picker. Not sure it's really worth it as anyone with a little common sense can figure that out.

**Comment by bhollis** [comment]
Is that a new behavior? We could fix it in the reducer.

**Comment by bhollis** [comment]
Looks like that's possible today and doesn't cause anything too crazy to happen (they cancel out)

**Pull request merged**